pkg/webhook/controlplane/ensurer.go

```diff
@@ -58,6 +59,18 @@ func (e *ensurer) InjectClient(client client.Client) error {
 	return nil
 }
 
+func computeCSIMigrationCompleteFeatureGate(cluster *extensionscontroller.Cluster) (string, error) {
+	k8sGreaterEqual121, err := versionutils.CompareVersions(cluster.Shoot.Spec.Kubernetes.Version, ">=", "1.21")
+	if err != nil {
+		return "", err
+	}
+
+	if k8sGreaterEqual121 {
+		return "InTreePluginAWSUnregister", nil
+	}
+	return "CSIMigrationAWSComplete", nil
```

Comment:
It would be better to have string constants here and reuse them across the package and tests.

Reply:
Yeah, generally I agree, however, in this case the only usage of those names are in this file, and this is even the only occurrence. For the tests I explicitly don't want to reuse the constants to be able to detect issues in case someone (accidentally?) changes a constant. Also, for the other provider's it wasn't done like this, so I suggest to proceed as is :)

Reply:
>  in this case the only usage of those names are in this file, and this is even the only occurrence

Agreed. That's why i kept it optional. 
/lgtm otherwise 👍